### PR TITLE
Searching dialog fix

### LINKED ISSUE #250: Switch networks doesnt have searching dialog
Unlike the add members to network

### LINKED ISSUE #300: Searching dialog disappears on rotation
When searching for a network to join the searching dialog disappears when you rotate

### PR DESCRIPTION
**Pull request by ejohnson44**
Fixes #250 and Fixes #300

Added in a searching dialog for switch networks button and fixes the rotation issues on all of the dialogs.

### CODE REVIEW

File: SoundStream/res/layout/fragment_network.xml [hunk 1]
```diff
@@ -15,9 +15,8 @@
     GNU General Public License for more details.
     
     You should have received a copy of the GNU General Public License
-    along with SoundStream.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (reidbaker) [inline SoundStream/res/layout/fragment_network.xml:18]: Did you know you changed this? Not a big deal it looks the same.

File: SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java [hunk 2]
```diff
@@ -131,6 +132,12 @@ public void run() {
         joinText.setText(String.format(
                          getString(R.string.join_network),
                          BluetoothUtils.getLocalBluetoothName()));
+        
+        if(savedInstanceState != null){
+            isSearching = savedInstanceState.getBoolean("isSearching");
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java]: "isSearching should be a class level string I suggest
`private final static String SEARCHING_KEY = "isSearching";`

File: SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java [hunk 3]
```diff
@@ -154,6 +161,12 @@ public void onDestroy() {
         super.onDestroy();
     }
     
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean("isSearching", isSearching);
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java]: "isSearching" should be a class level string

File: SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java [hunk 4]
```diff
@@ -200,19 +213,16 @@ public void onReceiveAction(Context context, Intent intent) {
                     if(joinView != null){
                         switch(mode){
                         case BluetoothAdapter.SCAN_MODE_NONE:
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java]: If SCAN_MODE_NONE and SCAN_MODE_CONNECTABLE have the same code why not have the first fall through to the second then break.

Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java]: After looking at this code I think an if statement might be better. 

``` java
if (mode == BluetoothAdapter.SCAN_MODE_NONE || mode == BluetoothAdapter.SCAN_MODE_CONNECTABLE {
    isSearching = false;
    setJoinToDefaultState();
} else if (mode == BluetoothAdapter.SCAN_MODE_CONNECTABLE_DISCOVERABLE) {
    isSearching = true;
    setJoinToSearchingState();
} else {
    Log.wtf(TAG, "Recieved scan mode changed with unknown mode");
}    
```

File: SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java [hunk 5]
```diff
@@ -223,6 +233,18 @@ public void onReceiveAction(Context context, Intent intent) {
...
+        joinView.setEnabled(true);
+        joinView.findViewById(R.id.searching).setVisibility(View.INVISIBLE);
+        joinView.setBackgroundColor(getResources().getColor(R.color.abs__background_holo_light));
+    }
+    
+    private void setJoinToSearchingState(){
+        joinView.setEnabled(false);
+        joinView.findViewById(R.id.searching).setVisibility(View.VISIBLE);
+        joinView.setBackgroundColor(getResources().getColor(R.color.gray));
+    }
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java:247]: We can reduce these methods by using the code below. 

``` java
    private void setJoinToDefaultState(){
        modifyJoinView(true, View.INVISIBLE, R.color.abs__background_holo_light));
    }

    private void setJoinToSearchingState(){
        modifyJoinView(false, View.VISIBLE, R.color.gray)
    }

   /**
   * @param enabled
   * @param visibility either {@link View.VISIBLE} or {@link View.INVISIBLE}
   * @param color from R.color
   */ 
   private void modifyJoinView(Boolean enabled, int visibility, int color) {
        joinView.setEnabled(enabled);
        joinView.findViewById(R.id.searching).setVisibility(visibility);
        joinView.setBackgroundColor(getResources().getColor(color));
   }
```

File: SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java [hunk 6]
```diff
@@ -169,6 +166,17 @@ public void run() {
                 });
             }
         });
+        
+        if(savedInstanceState != null){
+            isSearchingAdd = savedInstanceState.getBoolean("isSearchingAdd");
+            isSearchingJoinDifferent = savedInstanceState.getBoolean("isSearchingJoinDifferent");
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java]: "isSearchingJoinDifferent" and "isSearchingAdd" should be class level strings

File: SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java [hunk 7]
```diff
@@ -251,6 +259,17 @@ public void onResume(){
         getActivity().setTitle(getTitle());
     }
     
+    /* (non-Javadoc)
+     * @see android.support.v4.app.Fragment#onSaveInstanceState(android.os.Bundle)
+     */
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        // TODO Auto-generated method stub
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java]: remove

File: SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java [hunk 8]
```diff
@@ -307,6 +326,35 @@ public void onReceiveAction(Context context, Intent intent) {
...
             })
+            .addGlobalAction(BluetoothAdapter.ACTION_SCAN_MODE_CHANGED, new IBroadcastActionHandler() {
+                
+                @Override
+                public void onReceiveAction(Context context, Intent intent) {
+                    int mode = intent.getIntExtra(
+                            BluetoothAdapter.EXTRA_SCAN_MODE, BluetoothAdapter.SCAN_MODE_NONE);
+                    
+                    if(joinDifferentNetworkButton != null){
+                        switch(mode){
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java:335]: If you move to the if like I suggested above then change this to an if as well. If not then dont :smile:

File: SoundStream/res/layout/fragment_network.xml (hunk 1 above)
Comment (ejohnson44) [inline SoundStream/res/layout/fragment_network.xml:18]: Yeah it was something weird with formatting - spacing just changed a bit I think

File: SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java (hunk 4 above)
Comment (ejohnson44) [inline SoundStream/src/com/thelastcrusade/soundstream/components/ConnectFragment.java]: I'll use the fall through method - I like the if statement better, but I had gotten the bulk of this code from something Jesse had written using a switch, so for consistencies sake I want to leave it in

File: SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java [hunk 9]
```diff
@@ -367,6 +410,24 @@ private void disband() {
...
+        button.findViewById(R.id.image_background)
+            .setBackgroundColor(getActivity().getResources().getColor(R.color.white));
+        
+    }
+    
+    private void setButtonToSearchingState(LinearLayout button){
+        button.setEnabled(false);
+
+        //TODO: add a better indicator while discovering
+        //...seconds until discovery is finished, number of clients found, etc
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/components/NetworkFragment.java:426]: Do you have an idea for how to make accurate predictions for these values? It's a question that's outside of the scope of this pull request, but definitely one worth asking if we can improve our connection experience.

**Comment by dgreenhalgh** [comment]
:+1:

**Comment by reidbaker** [comment]
This looks good. Can I merge it?

**Comment by dgreenhalgh** [comment]
:shipit:

**Pull request merged**